### PR TITLE
FOUNDATIONS: Remove `User`

### LINKED ISSUE #112: FOUNDATIONS: Remove `User`

### PR DESCRIPTION
**Pull request by GulchexrasGithub**
Closes #112

### CODE REVIEW

File: Tarteeb.Api.Tests.Unit/Services/Foundations/Users/UserServiceTests.Logic.RemoveById.cs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+                    Times.Once);
+
+            this.storageBrokerMock.Verify(broker =>
+                broker.DeleteUserAsync(expectedInputUser),
+                    Times.Once);
+
+            this.storageBrokerMock.VerifyNoOtherCalls();
+            this.loggingBrokerMock.VerifyNoOtherCalls();
+            this.dateTimeBrokerMock.VerifyNoOtherCalls();
+
```
Comment (Javlonbek-dev) [inline Tarteeb.Api.Tests.Unit/Services/Foundations/Users/UserServiceTests.Logic.RemoveById.cs:56]: Remove Line

File: Tarteeb.Api/Services/Foundations/Users/UserService.cs [hunk 2]
```diff
@@ -52,5 +52,18 @@ public partial class UserService : IUserService
 
             return maybeUser;
         });
+
+        public ValueTask<User> RemoveUserByIdAsync(Guid userId) =>
+        TryCatch(async () =>
+        {
+            ValidateUserId(userId);
+
+            User someUser =
```
Comment (ElbekDeveloper) [inline Tarteeb.Api/Services/Foundations/Users/UserService.cs:61]: someUser emas maybeUser bo'ladi

File: Tarteeb.Api.Tests.Unit/Services/Foundations/Users/UserServiceTests.Validations.RemoveById.cs [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+            //when
+            ValueTask<User> removeUserByIdTask =
+                this.userService.RemoveUserByIdAsync(inputUserId);
+
+            UserValidationException actualUserValidationException =
+                await Assert.ThrowsAsync<UserValidationException>(removeUserByIdTask.AsTask);
+
+            //then
+            await Assert.ThrowsAsync<UserValidationException>(() =>
```
Comment (ElbekDeveloper) [inline Tarteeb.Api.Tests.Unit/Services/Foundations/Users/UserServiceTests.Validations.RemoveById.cs:75]: Bu kerak emas.

File: Tarteeb.Api.Tests.Unit/Services/Foundations/Users/UserServiceTests.Validations.RemoveById.cs [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+                removeUserByIdTask.AsTask());
```
Comment (ElbekDeveloper) [inline Tarteeb.Api.Tests.Unit/Services/Foundations/Users/UserServiceTests.Validations.RemoveById.cs:76]: O'rniga Should().beequ... bo'lishi kerak

**Review by ElbekDeveloper — CHANGES_REQUESTED** [review]

File: Tarteeb.Api.Tests.Unit/Services/Foundations/Users/UserServiceTests.Validations.RemoveById.cs [hunk 5]
```diff
@@ -0,0 +1,92 @@
+﻿using System;
```
Comment (jasurbekyusuf) [inline Tarteeb.Api.Tests.Unit/Services/Foundations/Users/UserServiceTests.Validations.RemoveById.cs:1]: Header?

**Review by jasurbekyusuf — COMMENTED** [review]
LGTM

**Review by Javlonbek-dev — APPROVED** [review]

File: Tarteeb.Api/Services/Foundations/Users/UserService.cs [hunk 6]
```diff
@@ -52,5 +52,18 @@ public partial class UserService : IUserService
...
             return maybeUser;
         });
+
+        public ValueTask<User> RemoveUserByIdAsync(Guid userId) =>
+        TryCatch(async () =>
+        {
+            ValidateUserId(userId);
+
+            User maybeUser =
+               await this.storageBroker.SelectUserByIdAsync(userId);
```
Comment (StudentofTATU) [inline Tarteeb.Api/Services/Foundations/Users/UserService.cs:62]: use TAB

**Review by StudentofTATU — APPROVED** [review]

**Review by ElbekDeveloper — APPROVED** [review]

**Pull request merged**